gen_api.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+$source = "php://stdin";
+if (isset($argv[1])) {
+    $source = $argv[1];
+}
+
+$input_lines = file($source);
+foreach ($input_lines as $line) {
+    // Strip AWS_EXTERN_C_BEGIN/END
+    if (preg_match('/AWS_EXTERN_C/', $line)) {
+        continue;
+    }
+
+    // Strip macros/includes
+    if (preg_match('/^#/', $line)) {
```

Comment:
this isn't actually stripping anything, right? it's just saying 
//if it matches the regex
// continue

Reply:
Correct! It's to strip the lines from the output.

Reply:
Oh right!  I spaced